keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -711,214 +647,324 @@ BucketAccess {
 
   Spec BucketAccessSpec {
     // BucketClaimName is the name of the BucketClaim.
+    // +required
     BucketClaimName string
 
-    // Protocol is the name of the Protocol 
-    // that this access credential is supposed to support
-    // If left empty, it will choose the protocol supported
-    // by the bucket. If the bucket supports multiple protocols,
-    // the end protocol is determined by the driver. 
-    // +optional
-    Protocol Protocol
-
-    // BucketAccessClassName is the name of the BucketAccessClass
+    // BucketAccessClassName is the name of the BucketAccessClass.
+    // +required
     BucketAccessClassName string
 
-    // CredentialsSecretName is the name of the secret that COSI should populate
-    // with the credentials. If a secret by this name already exists, then it is
-    // assumed that credentials have already been generated. It is not overridden.
-    // This secret is deleted when the BucketAccess is delted.
-    CredentialsSecretName string
-    
-    // ServiceAccountName is the name of the serviceAccount that COSI will map
-    // to the OSP service account when IAM styled authentication is specified
-	// +optional
-	ServiceAccountName string
+    // Protocol is the name of the Protocol that this access credential is expected to support.
+    // +required
+    Protocol Protocol
+
+    // AccessSecretName is the name of the Kubernetes secret that COSI should populate with access
+    // details and credentials.
+    // This secret is deleted when the BucketAccess is deleted.
+    // +required
+    AccessSecretName string
   }
 
   Status BucketAccessStatus {
-    // AccessGranted indicates the successful grant of privileges to access the bucket
-    AccessGranted bool
-    
-    // AccountID is the unique ID for the account in the OSP. It will be populated
-    // by the COSI sidecar once access has been successfully granted.
-    // +optional
+    // ReadyToUse indicates the successful grant of privileges to access the bucket.
+    ReadyToUse bool
+
+    // AccountID is the unique ID for the account in the OSP. It will be populated by the COSI
+    // sidecar once access has been successfully granted.
     AccountID string
+
+    // AccessedBucketName is the name of the Bucket resource that the access corresponds to. This is
+    // filled in by the Controller based on the BucketClaim so that the Sidecar knows what Bucket
+    // to allow access to for this BucketAccess.
+    // TODO: will have to update this to a list when 1-access:many-buckets support is added
```

Comment:
Do we want it for `v1alpha2` or next iteration? This cannot be added during `v1alpha2` after the KEP is finalized, as it will break the API.

Reply:
This PR doesn't contain all of the changes that will be part of v1alpha2, based on Xing's suggestion. 

https://github.com/kubernetes/enhancements/pull/5219 is a follow up that contains this change.

Reply:
Understood, thanks!